### DISCUSSION TITLE
How to use Visual Studio 2019 builds on a larger runner?

### DISCUSSION THREAD

**Discussion by mojca** [community/community#57338] [Actions]
### Select Topic Area

General

### Body

The default 2-core Windows runners are no longer fit to build our code (we are running out of memory on every second attempt).
We are now building code on `windows-latest-8-cores` for Visual Studio 2022, but I'm not sure how to build the code with VS 2019 as there is no `windows-2012-8-cores` available.

**Comment by myofficework000** [comment]
## To use Visual Studio 2019 builds on a larger runner, follow these steps:

1. **Set up a larger runner:** Ensure you have a machine with sufficient resources (CPU, RAM, storage) that has Visual Studio 2019 installed.

2. **Install dependencies:** Make sure all required dependencies for your project are installed on the larger runner, including SDKs, libraries, frameworks, or tools.

3. **Configure the build environment:** Set up the build environment on the larger runner by configuring environment variables, paths, and other project-specific settings.

4. **Copy the source code:** Transfer or clone your project's source code to the larger runner machine using a version control system (e.g., Git) or manual file transfer.

5. **Open the solution in Visual Studio:** Launch Visual Studio 2019 on the larger runner and open your project's solution file (.sln) from the copied source code. Ensure that the solution and project configurations are set up correctly.

6. **Build the project:** Use the Visual Studio build tools to compile and build your project. Select the desired build configuration (e.g., Debug or Release) and platform (e.g., x86 or x64) as required. Initiate the build from the Visual Studio menu, toolbar, or by using keyboard shortcuts (e.g., Ctrl+Shift+B).

7. **Monitor the build process:** Keep an eye on the build progress and check for any errors or warnings. Visual Studio will provide output messages and logs that can help identify and troubleshoot issues.

For more information and detailed instructions, you can refer to the [Visual Studio documentation](https://docs.microsoft.com/en-us/visualstudio/).

By following these steps, you should be able to leverage Visual Studio 2019 builds on a larger runner machine. Remember to customize the setup and configuration based on your project's specific requirements.

Feel free to reach out if you have any further questions or need assistance!

**Comment by mojca** [comment]
Which machine has Visual Studio 2019 installed? I only see these four available:
![image](https://github.com/community/community/assets/15606/17c16bcf-59a0-45b9-9270-ab2e4ae54588)
